backend/src/helpers.ts

```diff
@@ -0,0 +1,120 @@
+import { ethers } from 'hardhat';
+
+export async function loadYamlConfig(filePath: string): Promise<any> {
+  try {
+    const fileContents = await promises.readFile(filePath, 'utf8');
+    return yaml.load(fileContents);
+  } catch (error) {
+    console.error("Error reading YAML file", error);
+    return {};
+  }
+}
+
+export async function deployContract(hre: typeof import('hardhat'),
+                                      accounts: HardhatEthersSigner[],
+                                      contractName: string,
+                                      ...deployArgs: any[]) {
+    const { ethers } = hre;
+    const contractFactory = await ethers.getContractFactory(contractName, accounts[0]);
+    const contract = await contractFactory.deploy(...deployArgs);
+    await contract.waitForDeployment();
+    console.log(`${contractName} deployed at address: ${await contract.getAddress()}`);
+    return contract;
+  }
+
+export async function addQuestions(quizContract: any, questionsFile: string) {
+  const questions = JSON.parse(await promises.readFile(questionsFile, 'utf8'));
+  for (const question of questions) {
+    const tx = await quizContract.addQuestion(question.question, question.choices);
+    const receipt = await tx.wait();
+    console.log(`Added question: ${question.question}. Transaction hash: ${receipt!.hash}`);
+  }
+}
+
+export async function addCoupons(quizContract: any, couponsFile: string) {
+  const coupons = (await promises.readFile(couponsFile, 'utf8')).split('\n').filter(Boolean);
+  for (let i = 0; i < coupons.length; i += 20) {
+    const chunk = coupons.slice(i, i + 20);
+    const tx = await quizContract.addCoupons(chunk);
+    const receipt = await tx.wait();
+    console.log(`Added coupons: ${chunk}. Transaction hash: ${receipt!.hash}`);
+  }
+}
+
+export async function setNativeReward(hre: typeof import('hardhat'), quizContract: any, reward: string) {
+  const { ethers } = hre;
+  const tx = await quizContract.setPayoutReward(ethers.parseEther(reward));
+  const receipt = await tx.wait();
+  console.log(`Set reward to ${reward} ROSE. Transaction hash: ${receipt!.hash}`);
+}
+
+
+export async function addAllowMint(hre: typeof import('hardhat'), contract: any, address: string) {
+  const tx = await contract.addAllowMint(address);
+  await tx.wait();
+  console.log(`Address ${address} allowed to mint.`);
+}
+
+export async function removeAllowMint(hre: typeof import('hardhat'), contract: any, address: string) {
+  const tx = await contract.removeAllowMint(address);
+  await tx.wait();
+  console.log(`Address ${address} allowed to mint.`);
+}
+
+export async function setNftAddress(hre: typeof import('hardhat'), contract: any, nftAddress: string) {
+  const tx = await contract.setNftAddress(nftAddress);
+  await tx.wait();
+  console.log(`NFT address set to ${nftAddress}.`);
+}
+
+export async function storeNFT(hre: typeof import('hardhat'), quiz: any, nft: any, jsonMetadata: string) {
```

Comment:
I think we can remove nft instance from parameters right?

```suggestion
export async function storeNFT(hre: typeof import('hardhat'), quiz: any, jsonMetadata: string) {
```